src/plugins/discover/public/application/components/timechart_header/timechart_header.tsx

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useState } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiToolTip,
+  EuiText,
+  EuiSelect,
+  EuiIconTip,
+} from '@elastic/eui';
+import { I18nProvider, FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+
+export interface TimechartHeaderProps {
+  /**
+   * the query from date string
+   */
+  from: string;
+  /**
+   * the query to date string
+   */
+  to: string;
+  /**
+   * Interval Options
+   */
+  options: Array<{ display: string; val: string }>;
+  /**
+   * changes the interval
+   */
+  onChangeInterval: (interval: string) => void;
+  /**
+   * selected interval
+   */
+  stateInterval: string;
+  /**
+   * displays the scaled info of the interval
+   */
+  showScaledInfo: boolean | undefined;
+  /**
+   * scaled info description
+   */
+  bucketIntervalDescription: string;
+  /**
+   * bucket interval scale
+   */
+  bucketIntervalScale: number | undefined;
+}
+
+export function TimechartHeader({
+  from,
+  to,
+  options,
+  onChangeInterval,
+  stateInterval,
+  showScaledInfo,
+  bucketIntervalDescription,
+  bucketIntervalScale,
+}: TimechartHeaderProps) {
+  const [interval, setInterval] = useState(stateInterval);
+
+  const handleIntervalChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setInterval(e.target.value);
+    onChangeInterval(e.target.value);
+  };
```

Comment:
this internal state needs to be updated, if the `stateInterval` prop changes.

the stateInterval is part of the url, it's the **interval:ms** part.

http://localhost:5620/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))&_a=(columns:!(_source),filters:!(),index:ff959d40-b880-11e8-a6d9-e546fe2bba5f,interval:ms,query:(language:kuery,query:''),sort:!())

Reply:
You are right, I missed that!